android/capacitor/src/main/java/com/getcapacitor/Plugin.java

```diff
@@ -91,85 +95,111 @@ public Plugin() {
     public void load() {}
 
     /**
-     * Registers the permission launchers used by the {@link #requestPermissions(PluginCall)} plugin call and
-     * those defined on plugins
-     */
-    void initializePermissionLaunchers() {
-        basePermissionLauncher =
-            bridge
-                .getActivity()
-                .registerForActivityResult(
-                    new ActivityResultContracts.RequestMultiplePermissions(),
-                    permissions -> {
-                        PluginCall savedPermissionCall = bridge.getPermissionCall(handle.getId());
+     * Registers activity result launchers defined on plugins, used for permission requests and
+     * activities started for result.
+     */
+    void initializeActivityLaunchers() {
+        List<Method> pluginClassMethods = new ArrayList<>();
+        pluginClassMethods.addAll(Arrays.asList(getClass().getSuperclass().getDeclaredMethods()));
+        pluginClassMethods.addAll(Arrays.asList(getClass().getDeclaredMethods()));
```

Comment:
Something I just realized with this approach: we're assuming the super class is `Plugin`, which is _probably_ fine, but if someone extends another subclass, this may become difficult to debug. What do you think?

Reply:
I think to be safe I could rework this to scan up through the class hierarchy and stop at Plugin.class